app/code/Magento/Downloadable/Model/SampleRepository.php

```diff
@@ -308,8 +308,11 @@ protected function updateSample(
             $existingSample->setTitle($sample->getTitle());
         }
 
-        if ($sample->getSampleType() === 'file' && $sample->getSampleFileContent() === null) {
-            $sample->setSampleFile($existingSample->getSampleFile());
+        if ($sample->getSampleType() === 'file'
+            && $sample->getSampleFileContent() === null
+            && $sample->getSampleFileContent() !== null
```

Comment:
Such condition will always evaluate to false.

Did you mean `$sample->getSampleFile() !== null`?

Reply:
Hello @orlangur 
I have updated my code can review it.